src/object.js

```diff
@@ -122,7 +128,12 @@ function flush() {
 
     for (let k in props) {
       if (k.indexOf('.') === -1) { star = true; }
-      object.notify(k);
+      object.notify(
+        // strip '@' suffix if present
+        k.length > 1 && k[k.length - 1] === '@' && k[k.length - 2] !== '.' ?
```

Comment:
Could do `.endsWith('@.')` to favor clearer code.  Not sure about the performance but I don't think it should be too bad... and if it is I could fix that :)

Reply:
Here is what the code looks like using `endsWith`:

```javascript
k.length > 1 && k.endsWith('@') && !k.endsWith('.@') ?
  k.slice(0, k.length - 1) :
  k
```

Definitely looks nicer but my performance tests on node v8.4.0 says its 1.7-1.9 times slower.

Reply:
Minor, but if you make another commit I'd suggest making '@' and '.' constants with descriptive names for what they represent, assuming they're used elsewhere in the rest of the implementation.

Reply:
@burrows Thanks I actually misread the code,  I didn't see you were testing for `/[^\.]@$/` not `/.@$/`.  I'm guessing you tried regex to?

Sidenote: I'm actually surprised it's not slower :) It's currently implemented in CPP and not CSA (most of my CPP -> CSA has yielded >2x gains in V8)

Sidenote #2:  Keep in mind the performance profile of V8 in Node 8 and in the current Chrome is very different.

Reply:
Yep, tried regex too. Here is the code:

```javascript
const strs = ["foobars@", "foobars", "foobars.@", "@"];

const n = 1000000;

let t1;

console.log('testing regex:');
let results1 = [];
t1 = new Date;
for (var i = 0; i < n; i++) {
  for (var j = 0, m = strs.length; j < m; j++) {
    var str = strs[j];

    results1.push(str.replace(/([^.])@$/, '$1'));
  }
}
console.log('regex approach took:', (new Date) - t1);

console.log('testing cached regex:');
let results2 = [];
t1 = new Date;
let re = /([^.])@$/;
for (var i = 0; i < n; i++) {
  for (var j = 0, m = strs.length; j < m; j++) {
    var str = strs[j];

    results2.push(str.replace(re, '$1'));
  }
}
console.log('cached regex approach took:', (new Date) - t1);

console.log('testing index:');
let results3 = [];
t1 = new Date;
for (var i = 0; i < n; i++) {
  for (var j = 0, m = strs.length; j < m; j++) {
    var str = strs[j];

    results3.push(str.length > 1 && str[str.length - 1] === '@' && str[str.length - 2] !== '.' ? str.slice(0, str.length - 1) : str);
  }
}
console.log('index approach took:', (new Date) - t1);

console.log('testing endsWith:');
let results4 = [];
t1 = new Date;
for (var i = 0; i < n; i++) {
  for (var j = 0, m = strs.length; j < m; j++) {
    var str = strs[j];

    results4.push(str.length > 1 && str.endsWith('@') && !str.endsWith('.@') ? str.slice(0, str.length - 1) : str);
  }
}
console.log('endsWith approach took:', (new Date) - t1);
```

And the results:

```
testing regex:
regex approach took: 1117
testing cached regex:
cached regex approach took: 1086
testing index:
index approach took: 302
testing endsWith:
endsWith approach took: 538
``

Reply:
Well, just tried this in chrome and firefox and `endsWith` is actually a little faster in both browsers.

Reply:
Updated to use `endsWith` in 870805.